src/language-client-mutualization.ts

```diff
@@ -201,35 +203,42 @@ function bindClientToServer (
       }
     })))
 
-    const codelensRefreshSupport = clientCapabilities.workspace?.codeLens?.refreshSupport ?? false
-    const sendCodeLensRefresh = bindContext(() => {
-      if (codelensRefreshSupport) {
+    disposableCollection.push(languageClient.onCodeLensRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.codeLens?.refreshSupport ?? false) {
         clientConnection.sendRequest(CodeLensRefreshRequest.type).catch(error => {
           options.logger?.error('Unable to send Codelens token refresh to client', { error })
         })
       }
-    })
-
-    const semanticTokenRefreshSupport = clientCapabilities.workspace?.semanticTokens?.refreshSupport ?? false
-    const sendSemanticTokensRefresh = bindContext(() => {
-      if (semanticTokenRefreshSupport) {
+    })))
+    disposableCollection.push(languageClient.onSemanticTokensRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.semanticTokens?.refreshSupport ?? false) {
         clientConnection.sendRequest(SemanticTokensRefreshRequest.type).catch(error => {
           options.logger?.error('Unable to send semantic token refresh to client', { error })
         })
       }
-    })
-    const diagnosticsRefreshSupport = clientCapabilities.workspace?.diagnostics?.refreshSupport ?? false
-    const sendDiagnosticsRefresh = bindContext(() => {
-      if (diagnosticsRefreshSupport) {
+    })))
+    disposableCollection.push(languageClient.onDiagnosticsRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.diagnostics?.refreshSupport ?? false) {
         clientConnection.sendRequest(DiagnosticRefreshRequest.type).catch(error => {
           options.logger?.error('Unable to send Diagnostics refresh to client', { error })
         })
       }
-    })
+    })))
+    disposableCollection.push(languageClient.onInlayHintRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.inlayHint?.refreshSupport ?? false) {
```

Comment:
shouldn't we move this `if` around the `disposableCollection.push(...)` ? (Same for other `disposableCollection.push`)

Reply:
At the end, this check is useless as the server is already aware if the client supports it or not

=> c1679073c0b04b9d88f388c2ac6ce0002e9c0f74